src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+         call mpas_deallocate_scratch_field(sigmaminField, .true.)
+         !call mpas_deallocate_scratch_field(sigmamaxField, .true.)
+         call mpas_deallocate_scratch_field(alphaField, .true.)
+         !call mpas_deallocate_scratch_field(epsmaxField, .true.)
+         !call mpas_deallocate_scratch_field(nstarField, .true.)
+         !call mpas_deallocate_scratch_field(szeroField, .true.)
+         !call mpas_deallocate_scratch_field(sourceField, .true.)
+         !call mpas_deallocate_scratch_field(ddamagedtField, .true.)
+         call mpas_deallocate_scratch_field(forceField, .true.)
+         call mpas_deallocate_scratch_field(meanTField, .true.)
+         ! call mpas_deallocate_scratch_field(damageField, .true.)
+
+         block => block % next
+             
+      enddo
+
+   end subroutine calculate_damage 
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!    routine restore_damage
+!
+!> \author Tong Zhang
+!> \date   Oct. 2019
+!> \details restore the damage to the original value before advection if it exceeds some large number
+!-----------------------------------------------------------------------
+
+   subroutine restore_damage(domain, err)
+
+      !-----------------------------------------------------------------
+      ! input variables
+      !-----------------------------------------------------------------
+
+      !-----------------------------------------------------------------
+      ! input/output variables
+      !-----------------------------------------------------------------
+      type (domain_type), intent(inout) :: &
+         domain          !< Input/Output: domain object
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+      integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+      type (block_type), pointer :: block
+      type (mpas_pool_type), pointer :: geometryPool
+      type (mpas_pool_type), pointer :: meshPool
+      type (mpas_pool_type), pointer :: velocityPool
+      type (mpas_pool_type), pointer :: scratchPool
+      real(kind=RKIND), pointer :: config_damage_stiffness_min
+      logical, pointer :: config_damage_rheology_coupling
+      logical, pointer :: config_print_calving_info
+      real (kind=RKIND), dimension(:), pointer :: damage
+      real (kind=RKIND), dimension(:), pointer :: damage_max
+      real (kind=RKIND), dimension(:), pointer :: stiffnessFactor
+      real (kind=RKIND), pointer ::  &
+           config_ice_density,            & ! ice density
+           config_ocean_density             ! ocean density
+
+      real (kind=RKIND), save :: rhoi    ! ice density (kg m^{-3}), copied from config_ice_density
```

Comment:
these 2 variables are not used in this routine - please delete.

Reply:
deleted